docs/WALKTHROUGH.md

```diff
@@ -0,0 +1,394 @@
+- `Broker` Resource: a representation of the broker server in the service-catalog system
+    that indicates where a broker server is running
+- `ServiceClass`: a service offered by a particular service broker. `ServiceClass`es are created
+    in response to a `Broker` resource being submitted to the system
+- `Instance`: an instance of a `ServiceClass` provisioned by the `Broker` for that `ServiceClass`
+- `Binding`: a binding to an `Instance` which is manifested into a Kubernetes namespace
+
+These components are building blocks of the service catalog in Kubernetes from an
+API standpoint.
+
+----
+
+#### Note: accessing the service catalog
+
+Unfortunately, `kubectl` doesn't know how to speak to both the service catalog
+API server and the main Kubernetes API server without switching contexts or
+`kubeconfig` files.  For now, the best way to access the service catalog API
+server is via a dedicated `kubeconfig` file.  You can manage the kubeconfig in
+use within a directory using the `direnv` tool.
+
+### Creating kubeconfig
+Make sure to change to the service catalog directory and create the .kubeconfig
+file copying the following into your shell:
+
+console```
+cat << EOF > .kubeconfig
+apiVersion: v1
+clusters:
+- cluster:
+    server: http://<cluster ip>:80
+  name: service-catalog-cluster
+contexts:
+- context:
+    cluster: service-catalog-cluster
+  name: service-catalog-ctx
+current-context: service-catalog-ctx
+kind: Config
+preferences: {}
+EOF
+```
+
+The above `.kubeconfig` file used in combination with `direnv` is the simplest
+way to switch between the core API server and the service-catalog API server.
+An example of an .envrc file in the service catalog directory would look like:
+
+export KUBECONFIG=<path to service-catalog>/.kubeconfig
+
+Then kubectl commands executed from the service-catalog directory will
+now make requests to the correct API server.
+
+----
+
+Because we haven't created any resources in the service-catalog API server yet,
+`kubectl get` will return an empty list of resources:
+
+```console
+kubectl get brokers,serviceclasses,instances,bindings
+```
+
+Create a test-ns namespace while still executing Kubernetes API commands:
+```console
+kubectl create namespace test-ns
+```
+
+## Step 3 - Installing a UPS broker
+
+Service Catalog requires broker servers to operate properly. There is a User 
+Provided Service broker (UPS from now on), which allows consumption of existing
+services through the Service Catalog model. Just like any other broker, the
+UPS broker needs to be running somewhere before it can be added to the
+catalog. We need to deploy it first by using the
+[`ups-broker` Helm chart](../charts/ups-broker) into your cluster, just like
+you installed the catalog chart above. Run this simple command to install it with sensible
+defaults:
+
+```console
+helm install <path to service-catalog>/charts/ups-broker --name ups-broker --namespace ups-broker
```

Comment:
Apologies if this is wrong since I don't use direnv. Since we recommend using direnv earlier, maybe we should mention that you need to do the helm install outside of catalog API server context?